src/usr/local/kobocloud/getRemoteFile.sh

```diff
@@ -2,39 +2,41 @@
 
 linkLine="$1"
 localFile="$2"
-user="$3"    
+user="$3"
+outputFileTmp="/tmp/kobo-remote-file-tmp.log"
 
 #load config
 . `dirname $0`/config.sh
 
-
-if [ "$user" = "" ]; then
-    curlCommand=$CURL
-else
-    curlCommand="$CURL -u $user: "
-fi
-    
-#echo $curlCommand
-    
-echo "$linkLine -> $localFile"
-
-remoteSize=`$curlCommand -k -L --silent --head "$linkLine" | tr A-Z a-z | sed -n 's/^content-length\: \([0-9]*\).*/\1/p'`
-echo "Remote size: $remoteSize"
-if [ -f $localFile ]; then
-  localSize=`stat -c%s "$localFile"`
-else
-  localSize=0
+curlCommand="$CURL"
+if [ ! -z "$user" ] && [ "$user" != "-" ]; then
+    echo "User: $user"
+    curlCommand="$curlCommand -u $user: "
 fi
-if [ "$remoteSize" = "" ]; then
-  remoteSize=1
+
+echo "Download: "$curlCommand -k --silent -C - -L -o "$localFile" "$linkLine" -v
+
+$curlCommand -k --silent -C - -L -o "$localFile" "$linkLine" -v 2>$outputFileTmp
```

Comment:
Directly use `-C` to resume download if the file is present. This way we can still have _resumable_ downloads.

At the same time with `-v` we are getting headers information (response code) which we use later on to determine if the content downloaded is the file or a error page.